lib/hpke/test/common.cpp

```diff
@@ -5,6 +5,35 @@
 #include <doctest/doctest.h>
 #include <openssl/crypto.h>
 
+#if defined(WITH_OPENSSL3)
+#include <openssl/evp.h>
+#include <openssl/provider.h>
+
+static OSSL_PROVIDER* fips_prov = nullptr;
+
+static int
+FIPS_mode()
+{
+  if (OSSL_PROVIDER_available(nullptr, "fips") == 1) {
+    return 1;
+  }
+  return EVP_default_properties_is_fips_enabled(nullptr);
+}
```

Comment:
Rather than emulate the old OpenSSL FIPS API, I would pull the contents of `fips.cpp/h` into this file and its header, and make variants of `fips()` and `enable_fips()` for the two versions.  Where in the latter case, I mean either make a new function `enable_fips()` that varies between the versions, or just branch internally to `ensure_fips_if_required()`.

Reply:
Not a problem